designs/dynamic-bundle.md

```diff
@@ -0,0 +1,267 @@
+Dynamic Bundle
+===================
+
+## Overview
+Kubebuilder currently has an implementation of a `Bundle` which allows multiple plugins to be *bundled* together. The execution of the plugins within a `Bundle` follows a sequential order based on their definition. 
+
+An example of the current method:
+
+```go
+gov3Bundle, _ := plugin.NewBundle(golang.DefaultNameQualifier, plugin.Version{Number: 3},
+	kustomizecommonv1.Plugin{},
+	golangv3.Plugin{},
+)
+```
+
+In the above example a plugin `Bundle` is created that ensures that the `kustomize/v1` plugin is run before the `go/v3` plugin.
+ 
+With the addition of Phase 2 Plugins on the way, users using the `--plugins` flag to utilize external plugins will not get the benefits of using some default functionality that previous Kubebuilder users using the default `go/v3` plugin gained via the `kustomize/v1` plugin.
```

Comment:
In my understanding, `--plugins` field would ensure the ordering of plugins by running the plugins in order, irrespective of whether we are including internal plugins or external plugins in the plugin chain. So we will be able to ensure that the `kustomize/v1` plugin is run before the `go/v3` plugin, just like how it's run in the `Bundle` concept.

Can you clarify about what is the default functionality and why we are not able to reap those benefits when trying to run external plugins?

Reply:
This was also brought up by @camilamacedo86 previously and I answered here: https://github.com/kubernetes-sigs/kubebuilder/pull/2671#discussion_r865863397

If there are more questions after looking at that answer let me know!

Reply:
I have updated this section to reflect the actual intentions of this proposal